src/pdm/cli/commands/use.py

```diff
@@ -70,6 +84,25 @@ def version_matcher(py_version: PythonInfo) -> bool:
                 project.core.ui.info("Using the last selection, add '-i' to ignore it.")
                 return cached_python
 
+        if not python and not first and (auto_install_min or auto_install_max):
```

Comment:
Why should it be preferred than existing interpreters? Wouldn't it cause unnecessary installations when I run it more than once?

Reply:
I agree that this could be optimized to figure out, if the existing ones already meet "auto install" requirements (pdm installed cPython version). But then again, the installation function is already taking care of this (if already present then it is re-used to my understanding).
I explicitly wanted to use only pdm installed cPython interpreter for the "auto install" functionality to have an independent interpreter be used to avoid side-effects (conda environments etc.). This is also a good approach for CI/CD pipelines.
Since the original behavior is still present, a dedicated Python interpreter may be chosen or selected by user interaction if the "auto install" is not used.